tests/testthat/test_move_pieces.R

```diff
@@ -156,6 +211,150 @@ testthat::test_that(
 
         result <- validate_type_2_move(NULL, NULL, NULL, G1)
 
-        testthat::expect_true(!result)
+        testthat::expect_false(result)
     }
-)
\ No newline at end of file
+)
+
+
+testthat::test_that(
+    "Check validate_type_1_move() returns FALSE when b is not simple",
+    {
+        b <- igraph::graph_from_edgelist(
+            matrix(c(
+            c(1, 4),
+            c(1, 4),
+            c(4, 2),
+            c(2, 1),
+            c(1, 3)
+        ), ncol = 2, byrow = TRUE), directed = FALSE)
+
+        result <- validate_type_1_move(NULL, NULL, NULL, b)
+
+        testthat::expect_false(result)
+    }
+)
+
+
+testthat::test_that(
+    "Check validate_type_1_move() returns TRUE on valid example",
+    {
+        Gdir <- igraph::graph_from_edgelist(
+            matrix(c(
+            c(4,2)
+        ), ncol = 2, byrow = TRUE), directed = TRUE)
+
+        Gudir <- igraph::graph_from_edgelist(
+            matrix(c(
+            c(1,4),
+            c(2,3)
+        ), ncol = 2, byrow = TRUE), directed = FALSE)
+
+        r <- Gudir
+        
+        b <- igraph::graph_from_edgelist(
+            matrix(c(
+            c(1,2),
+            c(3,4)
+        ), ncol = 2, byrow = TRUE), directed = FALSE)
+
+        result <- validate_type_1_move(Gdir, Gudir, r, b)
+
+        testthat::expect_true(result)
+    }
+)
+
+
+testthat::test_that(
+    "Check validate_type_1_move() returns FALSE when b adds edge to directed component",
+    {
+        Gdir <- igraph::graph_from_edgelist(
+            matrix(c(
+            c(1,3),
+            c(1,4)
+        ), ncol = 2, byrow = TRUE), directed = TRUE)
+
+        Gudir <- igraph::graph_from_edgelist(
+            matrix(c(
+            c(1,2),
+            c(2,3)
+        ), ncol = 2, byrow = TRUE), directed = FALSE)
+
+        r <- Gudir
+        
+        b <- igraph::graph_from_edgelist(
+            matrix(c(
+            c(1,3),
+            c(3,4)
+        ), ncol = 2, byrow = TRUE), directed = FALSE)
+
+        result <- validate_type_1_move(Gdir, Gudir, r, b)
+
+        testthat::expect_false(result)
+    }
+)
+
+testthat::test_that(
+    "Test that generate_type_1_move() returns correct output for known example.",
+    {
+        gudir <- igraph::graph_from_edgelist(
+            matrix(c(
+                c(2, 6),
+                c(3, 5)
+            ), ncol = 2, byrow = TRUE), directed = FALSE)
+
+        gdir <- igraph::graph_from_edgelist(
+            matrix(c(
+                c(1, 6),
+                c(1, 2),
+                c(2, 6)
+                #c(2, 5),
+                #c(4, 3)
+            ), ncol = 2, byrow = TRUE), directed = TRUE)
+
+        b_expected <- igraph::graph_from_edgelist(
+            matrix(c(
+                c(2, 5),
+                c(3, 6)
+            ), ncol = 2, byrow = TRUE), directed = FALSE)
+
+        set.seed(1234)
+        result <- generate_type_1_move(gdir, gudir, NULL, NULL)
+
```

Comment:
For the sake of completeness, apply the move and check the degree sequences of the resulting graph with that in gudir.

Reply:
See #36